lib/pr.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from lib.sh import sh
+from lib.types import URL
+
+
+@dataclass(frozen=True)
+class PR:
+    url: URL
+
+    def is_closed(self) -> bool:
+        status = sh.stdout(
+            ("gh", "pr", "view", self.url, "--json", "state", "--jq", ".state")
+        )
+        return status in ("CLOSED", "MERGED")
```

Comment:
P3: With just one attribute, this class is better represented as a function:

```suggestion
    def is_closed(pr_url: URL) -> bool:
        status = sh.stdout(
            ("gh", "pr", "view", pr_url, "--json", "state", "--jq", ".state")
        )
        return status in ("CLOSED", "MERGED")
```